Sc24 draft

tilsche opened the pull request.

tilsche committed:
prototype for revised timing requirements

tilsche committed:
minor table adjustments

tilsche committed:
Change requirement after discussion

BranTran committed:
make consistent use of Level 1 instead of shortening to L1

tilsche committed:
Updating the sampling details.

This should fix #90, #88, #108.

I removed a lot of text that was realatively detaild, difficult to understand, and hardly possible to verify with realistic levels of measurement device specifications.

tilsche committed:
Use more acronyms.

Note: HPC is not consistently used for easier editing and fluffy rendering of hyperrefs.

tilsche committed:
Remove remaining reference to abbreviated Levels, follow-up on 2a4042d

tilsche committed:
remove reference to run the benchmark on all compute nodes.

To my knowledge, there is no such rule in the Top500,
many large systems actually do partial system runs,
even the submission form supports using a different number
of corse for the system and for running the benchmark.

tilsche committed:
Clarify that Level 1 and 2 do notnecessarily only measure a partial system.

This has actually caused some confusion in a heise.de press article recently,
so I'd like to be precise here.

tilsche committed:
Remove requirement to submit time series from text.

This is in accordance with the changes in the table 196fcca
and in line with the simplifications of #108.
It is still recommended to provide the data.

The reasoning was that this data is neither publicly available for
research purposes nor used for validation.

tilsche committed:
Replace old figure with static power consumption with new CLAIX example.

tilsche committed:
Remove references to energy from the checklist

This is in accordance with the updated requirements table that
allows either energy or high-resolution power for levels 2 and 3.

It also removes redundancy from the document, #101

tilsche committed:
Add reference to IEC standards, fixes #91

chriswasser committed:
Fix typo

chriswasser committed:
Clarify recommendation (and not requirement) of standardized PDUs

Previously, the wording of "Level 3 allows" caused confusion as many readers implied a requirement from that (as in "Level 3 only allows"). To clarify this recommendation, the wording was changed.

BranTran commented:
See [Comparison against Master](https://github.com/EEHPCWG/PowerMeasurementMethodology/compare/master...sc24?diff=split&w=)
Last Updated: Feb 4, 2025
* checklist.tex 
  * Line 53: Added "allow to" 
  * Line 100-108: Remove text 
    * gathering an additional set of "intermediate measurements", i.e. power averaged measurements at regular interval
    * references to measurement sections
  * Line 126-138: remove text
    * Level 3 energy measurement requirement
    * Reference discussing energy measurements/measurement/formatting
  * Line 141, 144: cosmetic changes to remove additional new lines
* definititions.tex
* intro.tex
* reporting.tex

Approvals: Feb 4, 2025
* [DONE] methodology.tex
  *  [APPROVED] stylistic changes
* [DONE] acronyms.tex
  * [APPROVED] Line 4+: Added HPC, HPL, PDU, EEHPC WG acronyms to the file
* checklist.tex
  * [APPROVED] Line 51: Remove line that workload must run on all compute nodes
  * [FIX] Line 52: Do NOT include new text "that runs the benchmark"

BranTran commented on text/checklist.tex:
```diff
@@ -48,9 +48,8 @@ \chapter{Checklist}
-The workload must run on all compute nodes of the system.
-Level~3 measures the power for the entire system.
-Levels~1 and~2 measure the power for a portion of the system and extrapolate a value for the entire system.
+Level~3 measures the power for the entire system that runs the benchmark.
```
For Level~3, all power used by the parts of a computer system and supporting components that participate in a workload must be included in the power measurement.

BranTran commented on text/checklist.tex:
```diff
@@ -48,9 +48,8 @@ \chapter{Checklist}
-Level~3 measures the power for the entire system.
-Levels~1 and~2 measure the power for a portion of the system and extrapolate a value for the entire system.
+Level~3 measures the power for the entire system that runs the benchmark.
+Levels~1 and~2 allow to measure the power for a portion of the system and extrapolate a value for the entire system.
```
Levels~1 and~2 allow measuring the power ...

BranTran commented on text/checklist.tex:
```diff
@@ -97,15 +96,6 @@ \chapter{Checklist}
 Refer to Section~\ref{sec:PAaTEM} for information on power-averaged measurements.
 The core phase is required to run for at least one minute.
 
```
[Line 91] Level~2 submissions include the average power of the core phase and the average power of the full run sampled at least once per second.\ref(sec:core_phase) This includes at least one idle power measurement.

Table Update (1b):
Average core phase
Average full run
Idle Power

BranTran commented on text/checklist.tex:
```diff
@@ -97,15 +96,6 @@ \chapter{Checklist}
-On top of these full measurements, Level~2 also requires a set of intermediate measurements in order to see how the power consumption varies over time.
-For this purpose, a series of power-averaged measurements of equal length at regular intervals must be submitted for the full run.
-These intervals must be short enough that at least \MinMeasurementsCorePhaseLTwoThree{} measurements are reported during the core phase of the workload.
-This series of measurements in total must cover the full run.
```
Approved removal from checklist.txt with revision*

Discuss adding encouraging mechanisms for releasing data in the body of the document (will be made for levels 2-3)
additional discussion raw data/preprocessed/aggregated data

BranTran commented on text/checklist.tex:
```diff
@@ -123,25 +113,10 @@ \chapter{Checklist}
-The measured energy is the last measured total energy within the core phase minus the first measured total energy within the core phase.
-The final power is calculated by dividing this energy by the elapsed time between these first and last energy readings.
-These last and first measurements in the core phase must be timed such that no more than a total of ten seconds (five each at begin and end) of the core phase are not covered by the total energy measurement.
-
```
Approved removal of energy definition with revisions*

In Section 3.1.2 Power Reading and Sampling, change the following headers:

"Reporting Average Power" becomes
-	Deriving Average Power Using Instantaneous Power Samples

"Using Continuous Integrated Energy" becomes
-	Deriving Average Power Using Continuous Integrated Energy
	
NOTE: Add a line in Section 3.1 to state that average power values are reported in kilowatts.
Comment: Submission page requires values provided in kilowatts.

BranTran commented on text/reporting.tex:
```diff
@@ -45,147 +45,149 @@ \subsection{Accuracy Requirements}
+The readout interval should be approximately constant for the duration of the observation.
+
+A power value read from a measurement device can represent the power at a specific moment (\emph{instantaneous power}) or an \emph{average power} for a certain interval (e.g. the last \SI{1}{\second}) or a \emph{running average} over a longer time window.
+However, it is often not clearly documented by devices, which is the case.
```
Remove new lines 62-63, Refer to Section 3.1.1 (Accuracy Requirement).

BranTran committed:
Update text based on EE HPC WG review

Acronym: fix for EE HPC WG
Checklist: update wording on Level 3 measurement to be full system
Checklist: update so only Level 2 allows to measure portion.
Checklist: Add wording on L2 summary to include at least one idle power measurement.
Reporting: Adding line about reporting in kilowatts
Reporting: Updating 3.1.2 subsubsections to be about deriving average power
Reporting: Updating table to call out average values and singular idle power
Git Workflow: updating upload artifact to v4

BranTran committed:
Update text for approved changes up to 06/03 meeting
- Correct EEHPC WG to EE HPC WG
- remove "power value can be x,y, or y" detail in reporting.tex
- change encouraged to highly recommended
- include "aggregate" to mentions of average power
- add variable names for energy and timestamp in the body of the text

chriswasser commented:
Notes from today's meeting on 2026-03-24:

Integrate the following changes:
- Where does the full system measurement requirement come into play?
    - Table 3.1: keep row with the same wording for all Levels for clarity (might be able to drop this later, but keep it for now)
    - „3.5 Aspect 2: Machine Fraction Instrumented“ repurpose into explanation of full system and explain our reasoning
        - mention that it is allowed to use a subset of nodes for example to improve HPL decomposition, resulting in overall better performance or energy efficiency
    - adjust examples section
    - machine fraction also mentioned in introduction —> check for consistency
- Explicitly mention in the methodology that power-optimized HPL runs are allowed (Top500 and Green500 runs can be different)
- Change „2 Checklist“ to a „Step-by-Step to a Level 1 Submission Guide“
    - Reordering seems sensible to have a linearized time view
    - Look for a Green500 submitter to provide examples for Level 1 submissions or also additional Level 2 and 3 submisisons (add link from step-by-step guide to examples chapter)
    - recommendation to only include a step-by-step for Level 1 based on previous feedback that the checklist feels redundant due to replicated information within the summaries of the different levels
    - step-by-step guide seems more beginner-friendly
    - also provide an additional step-by-step guide to enhance a Level 1 measurement to achieve the Level 2 requirements (and later a Level 2 to Level 3 transition)
- Move summary table to more prominent position before Introduction

Step-by-Step Draft:
1. Identify Measurement Devices (Install or Check already Installed ones)
2. Identify Measurement Accuracy of the Devices
3. Identify Measurement Timing Capability of the Devices
4. Identify included subsystems and assess the possibility to estimate the missing subsystems
5. Coordinate data collection / configure monitoring with HPL execution

Methodology Change Plan:
1. Move Summary Table to front
2. Remove Checklist
3. Provide Level 1 Step-by-Step

highlight differences between levels?!
- timing requirements
- meter accuracy
- subsystem estimation / measurement
- (also reporting requirements)

Not discussed at meeting, but still should be addressed:
- These last and first measurements in the core phase must be timed such that no more than a total of ten seconds (five each at the beginning and end) of the core phase are not covered by the total energy measurement. ==> no longer correct with updated requirements
- Add comment on „integrated required cooling devices“
- Remove need to measure the storage subsystem
- „Reporting Power Values“ description needs to be updated
- „Level 3 requires continuously integrated total energy measurements rather than power-averaged measurements. The readings must begin before the start of the run and extend to when it is finished.“ needs to be removed since we now also allow instantaneous power consumption samples in Level 3

BranTran committed:
Initial Rework that includes the following changes
- moved table summary into separate summary.tex
  + Update machine fraction and subsystems text
- moved checklist.tex to before the examples
  + Initial comment out of level 2 and level 3 information
- update text in section "Aspect-2: Machine Fraction Instrumented"

chriswasser committed:
feat: hide checklist, switch table position, update wording in table and Aspect 2

BranTran committed:
update: fix typo in table, emphasis all compute notes for clarity, update 2.5 machine fraction as suggested 4/21/26

chriswasser committed:
feat: use latexmk as build tool

chriswasser committed:
feat: upgrade versions of GitHub actions

chriswasser committed:
fix: GitHub action for new texlive-action version

chriswasser committed:
feat: upgrade more versions of GitHub actions

chriswasser committed:
docs: add comment for known bug in latexdiff

chriswasser committed:
feat: integrate draft for timing changes

BranTran committed:
Update:
-Remove unnecessary header for where Summary Table was originally placed
-Update Aspect 3 to align with table (L1 and L2 measure compute node and measure/estimate all subsystems)
-Update Aspect 3 to remove storage subsystem from list of subsystems
